### PR TITLE
Prevent intro height calcuation when closing

### PR DESCRIPTION
**Pull request by lemieux**
Long story short :  since the widget is scaling down to disappear, it was triggering the intro height recalculation, which updated the store, which triggered the conversation component to be updated, which caused the small animation glitch. This will now prevent the intro component to recompute its height when the widget is closed or closing (the widget enters the CLOSED state when starting the animation).

I was also tired by using a boolean to represent 3 possible states, so I made it a proper enum like it should have been from the start.

@alavers @dannytranlx @chloepouprom @spasiu @jugarrit

### CODE REVIEW

File: src/js/components/chat-input.jsx [hunk 1]
```diff
@@ -93,24 +93,22 @@ export class ChatInputComponent extends Component {
...
-                               onChange={ this.onChange }
-                               onFocus={ this.onFocus }
-                               value={ this.state.text }
-                               title={ ui.text.sendButtonText }></input>
-                    </div>
-                </form>
-                { sendButton }
-            </div>
-            );
+        return <div id='sk-footer'>
```
Comment (lemieux) [inline src/js/components/chat-input.jsx]: This change is completely unrelated :| ... I was bothered by the `()` around the markup... it's the only change there.

File: src/stylesheets/mixins.less [hunk 2]
```diff
@@ -41,3 +41,8 @@
         background-color: rgba(210, 210, 210, 0.97);
     }
 }
+
+.flex(@value) {
```
Comment (lemieux) [inline src/stylesheets/mixins.less]: Hmm, kind of unrelated :(, but I'll use it in another branch

File: src/js/components/introduction.jsx [hunk 3]
```diff
@@ -77,10 +81,12 @@ export class IntroductionComponent extends Component {
     }
 }
 
-export const Introduction = connect(({appState: {introHeight}}) => {
+export const Introduction = connect(({appState: {introHeight, widgetState}}) => {
+    console.log(widgetState, introHeight)
```
Comment (chloepouprom) [inline src/js/components/introduction.jsx]: console.log

**Comment by mspensieri** [comment]
👍

**Comment by chloepouprom** [comment]
👍

File: src/stylesheets/mixins.less (hunk 2 above)
Comment (dannytranlx) [inline src/stylesheets/mixins.less]: Jesus 😂
